### ISSUE TITLE
Feature: Specify certain columns as being right-aligned

### ISSUE THREAD

**Issue by petdance** [neilb/Text-Table-Tiny#6]
I'm doing a lot of tables with numeric data, and it would be good to have those right-aligned.  I don't think it should automatically be detected, but I could pass an arrayref of columns that should be right-aligned.

**Comment by petdance** [comment]
I'd be glad to look into writing this if it's something that you'd be interested in incorporating.

**Comment by neilb** [comment]
I've thought about this a few times. There are some other changes I want to make, so I'll start with some refactoring, and work this change in as well.

**Comment by neilb** [comment]
What I'm playing with at the moment is an `align` option:

    generate_table(rows => \@rows, align => 'llclr', header_row => 1);

Other options:

 * pass an array of strings for the alignment: `align => [qw/ l r r /]`
 * annotate columns in the first row, whether header or not, eg `"blah blah|"` for right align
 *  list column numbers for right aligning or left aligning

I decided that modifying cell data isn't a good approach, and it's clearest to specify the column alignment for all columns, if you want something other than left-alignment for all columns.

What do you think?

**Comment by neilb** [comment]
Thoughts on alignment options:

    l   left
    c  center
    r  right
    .  right align at decimal point

Though I probably won't implement the '.' option for the first version.

**Comment by petdance** [comment]
I think `align => [qw/ l r r /]` makes the most sense.  It matches how you specify columns.

**Comment by neilb** [comment]
Good point. And internally I was mapping the string to that, so it's easier this way.

I've written up the changes I've currently done: [blog post](http://neilb.org/2019/08/06/text-table-tiny-changes.html)

**Comment by neilb** [comment]
Finally (!) did a developer release for this, if you want to try it out?

**Comment by scottchiefbaker** [comment]
This works for me. 

You should consider a single option that aligns ALL the columns that way. If I have 8 columns, it might be nice to just say `align => 'center'` and have it apply to all of them.

**Comment by neilb** [comment]
I had the same thought about being able to specify a single alignment for all columns. I'll add that in the next developer release!

**Comment by neilb** [comment]
Now released in 1.00.

**Comment by petdance** [comment]
I finally got to using this.

```
+-------------+-------------+---------------------------+------------+---------------+------------+------------+
|      nbytes | table_name  | index_name                | index_type | distinct_keys | uniqueness | compressed |
+-------------+-------------+---------------------------+------------+---------------+------------+------------+
| 570,425,344 | SUGGESTIONS | SUGGESTIONS_PHRASE        |            |    10,331,875 | UNIQUE     |            |
| 203,423,744 | SUGGESTIONS | SUGGESTIONS_PK            |            |    10,331,875 | UNIQUE     |            |
|   3,145,728 | SUGGESTIONS | SUGGESTIONS_TW_SEARCHABLE | BITMAP     |             2 |            |            |
+-------------+-------------+---------------------------+------------+---------------+------------+------------+
```

Works swell.  Thank you.
